src/main/java/org/sagebionetworks/web/client/widget/table/v2/results/TablePageWidget.java

```diff
@@ -89,6 +91,12 @@ public void configure(QueryResultBundle bundle,
 		types = ColumnModelUtils.buildTypesForQueryResults(QueryBundleUtils.getSelectFromBundle(bundle), bundle.getColumnModels());
 		// setup the headers from the types
 		List<IsWidget> headers = new ArrayList<IsWidget>();
+		Set<String> sortedHeaders = new HashSet<String>();

```

Comment:
Let's use a map &lt;String, SortItem&gt; to avoid ever iterating over the sortList to find the right item.